templates/wrappers.hs

```diff
@@ -274,8 +285,29 @@ alexMonadScan = do
 #endif /* ALEX_MONAD_BYTESTRING */
   sc <- alexGetStartCode
   case alexScan inp__ sc of
-    AlexEOF -> alexEOF
-    AlexError ((AlexPn _ line column),_,_,_) -> alexError $ "lexical error at line " ++ (show line) ++ ", column " ++ (show column)
+    AlexEOF -> do
+      alexThrowErrors
+      alexEOF
+
+#if defined(ALEX_BASIC)
+    AlexError (p@(AlexPn _ line column), x, s) -> do
+#elif !defined(ALEX_MONAD_BYTESTRING)
+    AlexError (p@(AlexPn _ line column),x,y,s) -> do
+#else
+    AlexError (p@(AlexPn _ line column), x, s, y) -> do
+#endif
+
+      alexPushError $ "lexical error at line " ++ (show line) ++ ", column " ++ (show column)
+
+#if defined(ALEX_BASIC)
+      alexSetInput (p, x, tail s)
+#elif !defined(ALEX_MONAD_BYTESTRING)
+      alexSetInput (p, x, y, tail s)
+#else
+      alexSetInput (p, x, ByteString.tail s, y)
+#endif
```

Comment:
Maybe @simonmar would prefer another approach for shifting instead of using `tail` function. I'd be glad hearing any suggestion ;)